Remotion/Web/Development.WebTesting/HostingStrategies/AspNetCoreHostingStrategy.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Remotion.Utilities;
+using Remotion.Web.Development.WebTesting.Configuration;
+using Remotion.Web.Development.WebTesting.HostingStrategies.Configuration;
+
+namespace Remotion.Web.Development.WebTesting.HostingStrategies;
+
+/// <summary>
+/// Hosts the WebApplication using ASP.NET Core locally.
+/// </summary>
+public class AspNetCoreHostingStrategy : IHostingStrategy
+{
+  private readonly ITestSiteLayoutConfiguration _testSiteLayoutConfiguration;
+  private readonly int _port;
+  private readonly CancellationTokenSource _cancellationTokenSource;
+  private Task? _webServerTask;
+
+  /// <param name="testSiteLayoutConfiguration">The configuration of the layout of the used test site.</param>
+  /// <param name="port">Port to be used.</param>
+  public AspNetCoreHostingStrategy (ITestSiteLayoutConfiguration testSiteLayoutConfiguration, int port)
+  {
+    ArgumentUtility.CheckNotNull(nameof(testSiteLayoutConfiguration), testSiteLayoutConfiguration);
+    ArgumentUtility.CheckNotNull(nameof(port), port);
+    _testSiteLayoutConfiguration = testSiteLayoutConfiguration;
+    _port = port;
+
+    _cancellationTokenSource = new CancellationTokenSource();
+  }
+
+  /// <summary>
+  /// Constructor required for direct usage in <see cref="WebTestConfigurationSection"/>.
+  /// </summary>
+  /// <param name="testSiteLayoutConfiguration">The configuration of the layout of the used test site.</param>
+  /// <param name="properties">The configuration properties.</param>
+  [UsedImplicitly]
+  public AspNetCoreHostingStrategy ( ITestSiteLayoutConfiguration testSiteLayoutConfiguration, IReadOnlyDictionary<string, string> properties)
+      : this(
+          ArgumentUtility.CheckNotNull(nameof(testSiteLayoutConfiguration), testSiteLayoutConfiguration),
+          int.Parse(ArgumentUtility.CheckNotNull(nameof(properties), properties)["port"]!))
+  {
+  }
+
+  /// <inheritdoc/>
+  public void DeployAndStartWebApplication ()
+  {
+    if (_webServerTask != null)
+      return;
+
+    if (_testSiteLayoutConfiguration.ExecutablePath == _testSiteLayoutConfiguration.RootPath)
+    {
+      throw new ArgumentException(
+          "Executable path was not set while trying to host with ASP.NET core. Please add 'pathFromRootToExecutable' property to the testSiteLayout configuration parameter.");
+    }
+
+    var processWrapper = new AspNetCoreHostingProcessWrapper(
+        Path.Combine(_testSiteLayoutConfiguration.ExecutablePath),
+        _testSiteLayoutConfiguration.RootPath,
+        $"http://localhost:{_port}");
+
+    _webServerTask = processWrapper.RunWebServerAsync(_cancellationTokenSource.Token);
+  }
+
+  /// <inheritdoc/>
+  public void StopAndUndeployWebApplication ()
+  {
+    _cancellationTokenSource.Cancel();
+    if (_webServerTask != null && !_webServerTask.Wait(1000))
```

Comment:
I honestly think that 1000 is easier to read. Everyone knows that these timings are always in milliseconds, and when you start reading something with a possible waiting time, you already expect there to be 1000 by default anyways.

Reply:
Compromise: `TimeSpan.FromMilliSeconds(1000)` ?

Reasoning: I did a lot of work with Microsoft stuff and there's a complete disregard for consistent units in the configurations. You get mixture of seconds and minutes or bytes and kilobytes, weird defaults ,etc. This is why we have always tried to remember to add the unit to the name of the property in the configration so you see it without thinking about it. 

It's true that the normal expectation is things are specified in milliseconds, but I have learned that it's easier to be explict all the time rather than having to remember if this is an oversight or a special case or actually a correct setting. The longer read doesn't hurt since it's not complicated and you immediately know what's going on.